src/parquet/column/writer.h

```diff
@@ -164,11 +177,14 @@ inline void TypedColumnWriter<DType>::WriteBatch(int64_t num_values,
   }
 
   if (num_rows_ > expected_rows_) {
-    throw ParquetException("More rows were written in the column chunk then expected");
+    throw ParquetException("More rows were written in the column chunk than expected");
   }
 
   WriteValues(values_to_write, values);
 
+  auto stats = static_cast<TypedRowGroupStatistics<DType>*>(page_statistics_.get());
+  stats->Update(values, values_to_write, num_values - values_to_write);
```

Comment:
It makes sense indeed, I ran the benchmarks, and the performance hit is sometimes significant.